src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala

```diff
@@ -53,7 +63,31 @@ class CoreModuleImpl @Inject() (
   private[this] lazy val shutdownHookModule = ShutdownHooks()
   private[this] lazy val actorsModule = new ActorsModule(shutdownHookModule, actorSystem)
 
-  override lazy val leadershipModule = LeadershipModule(actorsModule.actorRefFactory, zk, leader)
+  override lazy val leadershipModule = LeadershipModule(actorsModule.actorRefFactory, zk, electionModule)
+  override lazy val electionModule = if (marathonConf.highlyAvailable()) {
+    new TwitterCommonsElectionService(
```

Comment:
I guess the ElectionModule is a good place for that. In my prototype I have a wrapper around the TwitterCommon and the Curator class which switches over and passes through the API.